### PR TITLE
Create fake type and shape information for input and output to bypass the check

### PR DESCRIPTION
**Pull request by bddppq**
onnx requires graph input and output to have type and shape info

### CODE REVIEW

File: onnx_caffe2/frontend.py [hunk 1]
```diff
@@ -289,9 +289,21 @@ def caffe2_net_to_onnx_graph(net_def):
     graph_def.name = net_def.name
     # This is a mapping from Caffe2 names to ONNX names
     name_map = NameMap()
-    graph_def.input.extend(map(name_map.rename, net_def.external_input))
+    # TODO: This is cheating. Invent some magic to infer the types and
+    # shapes of graph input and output
+    graph_def.input.extend(
```
Comment (jerryzh168) [inline onnx_caffe2/frontend.py]: Should we ask users for these shape and type info for caffe2_net?

Comment (bddppq) [reply onnx_caffe2/frontend.py]: Yep, that's the straightforward (also my most preferred) way.

There is also discussion to export Caffe2 models in [MetaNetDef](https://github.com/caffe2/caffe2/blob/master/caffe2/proto/metanet.proto#L35) format.

I quickly post this hacky version to unblock pytorch side doing their fixes for the breaking change.

File: tests/caffe2_ref_test.py [hunk 2]
```diff
@@ -27,39 +27,42 @@
...
             node_def, {"X": X})
+        np.testing.assert_almost_equal(output.X, Y_ref)
+
         graph_def = make_graph(
             [node_def],
             name="test",
-            inputs=["X"],
-            outputs=["X", "Y"])
-        Y_ref = np.clip(X, 0, np.inf)
+            inputs=[make_tensor_value_info("X", onnx.TensorProto.FLOAT, [1, 2])],
```
Comment (jerryzh168) [inline tests/caffe2_ref_test.py]: Should this be [3, 2]?

Comment (bddppq) [reply tests/caffe2_ref_test.py]: Oh yeah, good catch :-) Will fix this.

**Comment by bddppq** [comment]
@jerryzh168 IIUC the BlobMap is mapping from blob name to the raw bytes representation of the blob value, kinda like `raw_data` field in onnx.TensorProto. (@dzhulgakov correct me if I'm wrong).

File: onnx_caffe2/frontend.py (hunk 1 above)
Comment (ezyang) [reply onnx_caffe2/frontend.py]: In the future, I'd recommend emitting a warning when you exercise these code paths; it serves as a clear reminder that something *extremely* dodgy is happening until it gets excised ;)

Comment (bddppq) [reply onnx_caffe2/frontend.py]: Yeah, will improve the error/warning message to help debugging in the future. For this one, I will commit a fix to clear this dodgy thing :-).

**Pull request merged**